src/Microsoft.OpenApi.Readers/V2/OpenApiV2VersionService.cs

```diff
@@ -66,6 +66,14 @@ public bool TryLoadReference(ParsingContext context, OpenApiReference reference,
 
             var node = context.RootNode.Find(jsonPointer);
 
+            if (node == null)
+            {
+                throw new OpenApiException(
+                    string.Format(
+                        SRResource.JsonPointerCannotBeResolved,
+                        jsonPointer));
+            }
+
```

Comment:
Maybe once we consolidate the error classes we can just throw this exception from inside the Find.  Is there somewhere where we need the response to be null on fail?

Reply:
Throwing the error here allows us to control what to do with json pointers that we cannot resolve. At least, we can leave them as a reference object (and reporting it) rather than always throwing an error and stopping the entire parsing process.

This probably no longer applies with the new reference resolving method.

---
In reply to: [165094630](https://github.com/Microsoft/OpenAPI.NET/pull/197#discussion_r165094630) [](ancestors = 165094630)